src/components/modals/ManageProjectAddresses/ManageProjectAddressesModal.tsx

```diff
@@ -16,68 +17,56 @@ import config from '@/configuration';
 import { NetworkWalletAddress } from './NetworkWalletAddress';
 import { AddNewAddress } from './AddNewAddress';
 import { getChainName } from '@/lib/network';
+import { NonEVMChain } from '@/types/config';
 import type { IModal } from '@/types/common';
 
-const networksConfig = config.EVM_NETWORKS_CONFIG;
-const networkIds = Object.keys(networksConfig).map(Number);
+const { CHAINS } = config;
 
-interface IManageProjectAddressesModal extends IModal {
+interface IModalProps extends IModal {
 	project: IProject;
 	setProjects: Dispatch<SetStateAction<IProject[]>>;
 }
 
-export const ManageProjectAddressesModal: FC<IManageProjectAddressesModal> = ({
-	project,
-	setShowModal,
-	setProjects,
-}) => {
-	const [selectedWallet, setSelectedWallet] = useState<IWalletAddress>();
-	const [addresses, setAddresses] = useState<IWalletAddress[]>([]);
+export const ManageProjectAddressesModal: FC<IModalProps> = props => {
+	const { project, setShowModal, setProjects } = props;
+	const [selectedChain, setSelectedChain] = useState<Chain | NonEVMChain>();
+	const [addresses, setAddresses] = useState<IWalletAddress[]>(
+		project.addresses || [],
+	);
 	const { isAnimating, closeModal } = useModalAnimation(setShowModal);
 	const { formatMessage } = useIntl();
-
-	useEffect(() => {
-		let WalletAddr: { [key: number]: IWalletAddress } = {};
-		networkIds.forEach(networkId => {
-			WalletAddr[networkId] = { networkId };
-		});
-		const { addresses } = project;
-		if (!addresses) return;
-		for (let i = 0; i < addresses.length; i++) {
-			const address = addresses[i];
-			if (address.networkId) {
-				WalletAddr[address.networkId] = address;
-			}
-		}
-		setAddresses(Object.values(WalletAddr));
-	}, [project]);
+	const selectedChainType =
+		selectedChain && 'chainType' in selectedChain
+			? selectedChain.chainType
+			: undefined;
```

Comment:
Like above
How about replacing it with
`const selectedChainType = selectedChain.chainType` it would have same result

Reply:
just like above I can't use this suggestion because "chainType" is not defined in "Chain" and I get typescript error

Reply:
Got it, thanks